packages-exp/auth-compat-exp/src/user.ts

```diff
@@ -47,7 +47,7 @@ export class User extends impl.UserImpl implements compat.User {
     credential: compat.AuthCredential
   ): Promise<compat.UserCredential> {
     return convertCredential(
-      this.auth,
+      (this.auth as unknown) as externs.Auth,
```

Comment:
@avolkovi Are these castings safe? I think you mentioned we never use the public APIs internally.

Reply:
there are three methods that have a changed signature in the new API vs the old one, we don't ever call those directly internally so it should be safe, I do a similar cast in auth-compat-exp